### DISCUSSION TITLE
给profile挂件中的文章计数器添加单独开关？

### DISCUSSION THREAD

**Discussion by lamya3** [ppoffice/hexo-theme-icarus#985] [Ideas]
觉得某些页面中profile挂件文章计数器用途不大，但是`_config.icarus.yml`中没有相关的配置选项，于是索性把`.../layout/widget/profile.jsx`里面这段计数器代码删掉实现关闭：
<details>
<font color="#FF0000"><summary>删掉的代码</summary></font>

```html
                <nav class="level is-mobile">
                    <div class="level-item has-text-centered is-marginless">
                        <div>
                            <p class="heading">{counter.post.title}</p>
                            <a href={counter.post.url}>
                                <p class="title">{counter.post.count}</p>
                            </a>
                        </div>
                    </div>
                    <div class="level-item has-text-centered is-marginless">
                        <div>
                            <p class="heading">{counter.category.title}</p>
                            <a href={counter.category.url}>
                                <p class="title">{counter.category.count}</p>
                            </a>
                        </div>
                    </div>
                    <div class="level-item has-text-centered is-marginless">
                        <div>
                            <p class="heading">{counter.tag.title}</p>
                            <a href={counter.tag.url}>
                                <p class="title">{counter.tag.count}</p>
                            </a>
                        </div>
                    </div>
                </nav>
```

</details>

![修改效果](https://s2.loli.net/2021/12/11/NGXvi9loDcICtWP.png)

但是在想把计数器移动到recent_posts或者archives的挂件的顶部这些地方，好像没有什么地方可以插入进去.....`.../layout/widget/`目录也只有一个profile.jsx。实际上可以通过临时修改代码做到这种效果吗？

或者提议一下增加类似这样的功能？：
- 给profile挂件里面的文章计数器部分增加开关配置项？使得可以通过配置的优先级控制不同页面是否显示计数器；
- 或者单独做一个挂件用于显示文章计数器？
- 或者可以把计数器移动到recent_posts或者archives这些挂件的上方？
- ...或者更有趣的点子

**Comment by ppoffice** [comment]
https://ppoffice.github.io/hexo-theme-icarus/uncategorized/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98/#%E5%B8%83%E5%B1%80

**Reply by lamya3** [reply]
:pray:感谢！
我以为把删掉的代码粘贴到`recent_posts.jsx`上面就完事了，但是渲染的时候报错才发现把计数器忘了:
```bash
ERROR { err:ReferenceError: counter is not defined...
```
...咱没怎么学过html，也没见过jsx，不太懂计数器添加的方法...靠着报错提示东墙补西墙，误打误撞好像行了
![QQ截图20211211171345.png](https://s2.loli.net/2021/12/11/3m1zwy4QSliN7xG.png)
但是我不知道这么改有没有问题，至少功能上好像没毛病。可以的话看看https://github.com/polarised969/hexo-component-inferno/commit/c226acf8d8f5389a483c478d39985c077eb18ca3 给些意见？:pray:再次感谢！
